script/github-actions/check-deployability.js

```diff
@@ -0,0 +1,174 @@
+  const params = {
+    owner,
+    repo,
+    workflow_id,
+    branch: 'main',
+    status: 'in_progress',
+  };
+
+  return octokit.rest.actions
+    .listWorkflowRuns(params)
+    .then(response => {
+      if (response.status !== 200) {
+        throw new Error(
+          `Response ${response.status} from ${response.url}. Aborting.`,
+        );
+      }
+      return response.data;
+    })
+    .then(({ workflow_runs }) => {
+      return workflow_runs;
+    });
+};
+
+/**
+ * Gets the commit hash of the last full deploy of an environment.
+ *
+ * @param {string} env - Name of environment
+ * @returns {string} Commit hash of the latest full deploy.
+ */
+const getLastFullDeployCommit = async env => {
+  const envBucketUrl = BUCKETS[env];
+  const buildTextUrl = new URL(path.join(envBucketUrl, 'BUILD.txt'));
+
+  const response = await fetch(buildTextUrl);
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch ${buildTextUrl}.\n${response.statusText}`);
+  }
+
+  const buildTextFile = await response.text();
+  const lines = buildTextFile.split('\n').filter(line => line);
+
+  return lines[6]?.slice(4);
+};
+
+/**
+ * Checks whether the first commit is an ancestor of the second commit.
+ *
+ * @param {string} commitA - Possible ancestor's commit hash
+ * @param {string} commitB - Possible descendant's commit hash
+ * @returns {Boolean} Returns true if the first commit is an ancestor of the second.
+ */
+const isAncestor = (commitA, commitB) => {
+  const exitCode = runCommandSync(
+    `git merge-base --is-ancestor ${commitA} ${commitB}`,
+  );
+
+  if (![0, 1].includes(exitCode)) {
+    throw new Error(
+      `'git merge-base --is-ancestor' exited with an unsuccessful code (${exitCode}).`,
+    );
+  }
+
+  return commitA !== commitB && exitCode === 0;
+};
+
+/**
+ * Checks if the GITHUB_SHA is ahead of the last full deploy of an environment.
+ *
+ * @param {string} env - Environment name
+ * @returns {Boolean} Returns true if GITHUB_SHA is ahead of the last full deploy.
+ */
+const isAheadOfLastFullDeploy = async env => {
+  const lastFullDeployCommit = await getLastFullDeployCommit(env);
+  return isAncestor(GITHUB_SHA, lastFullDeployCommit);
+};
+
+/**
+ * Determines if the GITHUB_SHA can be deployed to non production environments.
+ *
+ * @param {string} env - Environment name
+ * @returns {Boolean} Whether or not the GITHUB_SHA can be deployed to the environment.
+ */
+const checkDeployability = async env => {
+  if (!(await isAheadOfLastFullDeploy(env))) return false;
+
+  const inProgressWorkflowRuns = await getInProgressWorkflowRuns(
+    'continuous-integration.yml',
+  );
+  if (inProgressWorkflowRuns.length === 1) return true;
+
+  const previousCommitsInProgress = inProgressWorkflowRuns.find(workflowRun =>
+    isAncestor(workflowRun.head_sha, GITHUB_SHA),
+  );
+
+  if (!previousCommitsInProgress) return true;
+
+  const timeout = 5; // Number of minutes to wait before checking again
```

Comment:
Potentially this could be defined up at the top as an uppercase constant, so it's easier to find/change

Reply:
Yeah that's a good point, especially since we may be tweaking this often at first to get the right timeout